### ISSUE TITLE
[java] Error while processing in pmd 5.8.0

### ISSUE THREAD

**Issue by Nimfadora** [pmd/pmd#473]
**Rule Set**: N/A

**Description**: 

When I'm running pmd 5.8.0 on [checkstyle](https://github.com/checkstyle/checkstyle) project code, I'm getting following error:

```
[WARNING] There are 1 PMD processing errors:
[WARNING] ...\src\test\java\com\puppycrawl\tools\checkstyle\internal\XdocsPagesTest.java: Error while processing ...\src\test\java\com\puppycrawl\tools\checkstyle\internal\XdocsPagesTest.java
```
Link to the [XdocsPagesTest.java](https://github.com/checkstyle/checkstyle/blob/master/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java)

**Running PMD through**: Apache Maven 3.3.9

**Comment by jsotuyod** [comment]
@Nimfadora thanks for the report.

I'll check it right away, it may be a duplicate for #471 but just in case.

5.8.1 is scheduled for the end of this week, expect this fixed by then.

By the way, just yesterday's night I was going through a commit from yourself on Checkstyle and was meaning to ask you about [this](https://github.com/checkstyle/checkstyle/commit/f884cb3a269d87f3431cbb6d6030850ca2b0e8fb#diff-b8fa8c3b1a52e2f743babec46ae8abd9R16). Could you please create an issue with a working sample of such false positives? I'd rather fix them than have you suppress them.

**Comment by jsotuyod** [comment]
@Nimfadora I'm not able to reproduce the issue. I cloned Checkstyle, I changed the pom to use PMD 5.8.0, I run `mvn pmd:pmd` and I get the following report xml:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<pmd version="5.8.0" timestamp="2017-06-27T14:00:14.311">
<file name="redacted/checkstyle/src/main/java/com/puppycrawl/tools/checkstyle/PropertyCacheFile.java">
<violation beginline="221" endline="221" begincolumn="27" endcolumn="78" rule="UnusedPrivateMethod" ruleset="Unused Code" package="com.puppycrawl.tools.checkstyle" class="PropertyCacheFile" method="getHashCodeBasedOnObjectContent" externalInfoUrl="https://pmd.github.io/pmd-5.8.0/pmd-java/rules/java/unusedcode.html#UnusedPrivateMethod" priority="3">
Avoid unused private methods such as 'getHashCodeBasedOnObjectContent(Serializable)'.
</violation>
</file>
<file name="redacted/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/checks/indentation/IndentationCheckTest.java">
<violation beginline="147" endline="148" begincolumn="18" endcolumn="39" rule="UnusedPrivateMethod" ruleset="Unused Code" package="com.puppycrawl.tools.checkstyle.checks.indentation" class="IndentationCheckTest" method="verifyWarns" externalInfoUrl="https://pmd.github.io/pmd-5.8.0/pmd-java/rules/java/unusedcode.html#UnusedPrivateMethod" priority="3">
Avoid unused private methods such as 'verifyWarns(Configuration,String,String...)'.
</violation>
</file>
<file name="redacted/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWarningsFilterTest.java">
<violation beginline="108" endline="109" begincolumn="18" endcolumn="34" rule="UnusedPrivateMethod" ruleset="Unused Code" package="com.puppycrawl.tools.checkstyle.filters" class="SuppressWarningsFilterTest" method="verifySuppressed" externalInfoUrl="https://pmd.github.io/pmd-5.8.0/pmd-java/rules/java/unusedcode.html#UnusedPrivateMethod" priority="3">
Avoid unused private methods such as 'verifySuppressed(Configuration,String...)'.
</violation>
</file>
<file name="redacted/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyCommentFilterTest.java">
<violation beginline="201" endline="202" begincolumn="18" endcolumn="33" rule="UnusedPrivateMethod" ruleset="Unused Code" package="com.puppycrawl.tools.checkstyle.filters" class="SuppressWithNearbyCommentFilterTest" method="verifySuppressed" externalInfoUrl="https://pmd.github.io/pmd-5.8.0/pmd-java/rules/java/unusedcode.html#UnusedPrivateMethod" priority="3">
Avoid unused private methods such as 'verifySuppressed(Configuration,String...)'.
</violation>
</file>
</pmd>
```

Please, tell me what you are doing differently, or run `mvn` with `-X` so that debug output is printed and you may find a complete stacktrace.

**Comment by romani** [comment]
on our master:
```
$ git diff
diff --git a/pom.xml b/pom.xml
index 43bcb62..0b0e90d 100644
--- a/pom.xml
+++ b/pom.xml
@@ -201,7 +201,7 @@
     <maven.site.plugin.version>3.6</maven.site.plugin.version>
     <maven.findbugs.plugin.version>3.0.4</maven.findbugs.plugin.version>
     <maven.pmd.plugin.version>3.8</maven.pmd.plugin.version>
-    <pmd.version>5.7.0</pmd.version>
+    <pmd.version>5.8.0</pmd.version>
     <maven.jacoco.plugin.version>0.7.9</maven.jacoco.plugin.version>
     <maven.checkstyle.plugin.version>2.17</maven.checkstyle.plugin.version>
     <maven.sevntu.checkstyle.plugin.version>1.24.0</maven.sevntu.checkstyle.plugin.version>

$ mvn pmd:check
[INFO] Scanning for projects...
[INFO] Inspecting build with total of 1 modules...
[INFO] Installing Nexus Staging features:
[INFO]   ... total of 1 executions of maven-deploy-plugin replaced with nexus-staging-maven-plugin
[INFO]                                                                         
[INFO] ------------------------------------------------------------------------
[INFO] Building checkstyle 8.0-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] >>> maven-pmd-plugin:3.8:check (default-cli) > :pmd @ checkstyle >>>
Downloading: http://mvn.reveredata.com/maven/junit/junit-dep/maven-metadata.xml
Downloading: https://repo.maven.apache.org/maven2/junit/junit-dep/maven-metadata.xml
Downloading: https://oss.sonatype.org/content/repositories/snapshots/junit/junit-dep/maven-metadata.xml
Downloaded: https://repo.maven.apache.org/maven2/junit/junit-dep/maven-metadata.xml (563 B at 2.4 KB/sec)
Downloading: http://mvn.reveredata.com/maven/org/eclipse/jgit/org.eclipse.jgit/4.8.0.201706111038-r/org.eclipse.jgit-4.8.0.201706111038-r.pom
Downloading: https://repo.maven.apache.org/maven2/org/eclipse/jgit/org.eclipse.jgit/4.8.0.201706111038-r/org.eclipse.jgit-4.8.0.201706111038-r.pom
Downloaded: https://repo.maven.apache.org/maven2/org/eclipse/jgit/org.eclipse.jgit/4.8.0.201706111038-r/org.eclipse.jgit-4.8.0.201706111038-r.pom (10 KB at 294.5 KB/sec)
Downloading: http://mvn.reveredata.com/maven/org/eclipse/jgit/org.eclipse.jgit-parent/4.8.0.201706111038-r/org.eclipse.jgit-parent-4.8.0.201706111038-r.pom
Downloading: https://repo.maven.apache.org/maven2/org/eclipse/jgit/org.eclipse.jgit-parent/4.8.0.201706111038-r/org.eclipse.jgit-parent-4.8.0.201706111038-r.pom
Downloaded: https://repo.maven.apache.org/maven2/org/eclipse/jgit/org.eclipse.jgit-parent/4.8.0.201706111038-r/org.eclipse.jgit-parent-4.8.0.201706111038-r.pom (29 KB at 658.9 KB/sec)
Downloading: http://mvn.reveredata.com/maven/org/eclipse/jgit/org.eclipse.jgit/4.8.0.201706111038-r/org.eclipse.jgit-4.8.0.201706111038-r.jar
Downloading: https://repo.maven.apache.org/maven2/org/eclipse/jgit/org.eclipse.jgit/4.8.0.201706111038-r/org.eclipse.jgit-4.8.0.201706111038-r.jar
Downloaded: https://repo.maven.apache.org/maven2/org/eclipse/jgit/org.eclipse.jgit/4.8.0.201706111038-r/org.eclipse.jgit-4.8.0.201706111038-r.jar (2417 KB at 7846.5 KB/sec)
[INFO] 
[INFO] --- maven-pmd-plugin:3.8:pmd (pmd) @ checkstyle ---
Downloading: http://mvn.reveredata.com/maven/net/sourceforge/pmd/pmd-java/5.8.0/pmd-java-5.8.0.pom
Downloading: https://oss.sonatype.org/content/repositories/codehaus-snapshots/net/sourceforge/pmd/pmd-java/5.8.0/pmd-java-5.8.0.pom
Downloading: http://sevntu-checkstyle.github.io/sevntu.checkstyle/maven2/net/sourceforge/pmd/pmd-java/5.8.0/pmd-java-5.8.0.pom
Downloading: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-java/5.8.0/pmd-java-5.8.0.pom
Downloaded: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-java/5.8.0/pmd-java-5.8.0.pom (7 KB at 221.2 KB/sec)
Downloading: http://mvn.reveredata.com/maven/net/sourceforge/pmd/pmd/5.8.0/pmd-5.8.0.pom
Downloading: https://oss.sonatype.org/content/repositories/codehaus-snapshots/net/sourceforge/pmd/pmd/5.8.0/pmd-5.8.0.pom
Downloading: http://sevntu-checkstyle.github.io/sevntu.checkstyle/maven2/net/sourceforge/pmd/pmd/5.8.0/pmd-5.8.0.pom
Downloading: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd/5.8.0/pmd-5.8.0.pom
Downloaded: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd/5.8.0/pmd-5.8.0.pom (41 KB at 1362.2 KB/sec)
Downloading: http://mvn.reveredata.com/maven/net/sourceforge/pmd/pmd-core/5.8.0/pmd-core-5.8.0.pom
Downloading: https://oss.sonatype.org/content/repositories/codehaus-snapshots/net/sourceforge/pmd/pmd-core/5.8.0/pmd-core-5.8.0.pom
Downloading: http://sevntu-checkstyle.github.io/sevntu.checkstyle/maven2/net/sourceforge/pmd/pmd-core/5.8.0/pmd-core-5.8.0.pom
Downloading: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-core/5.8.0/pmd-core-5.8.0.pom
Downloaded: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-core/5.8.0/pmd-core-5.8.0.pom (6 KB at 231.3 KB/sec)
Downloading: http://mvn.reveredata.com/maven/net/sourceforge/pmd/pmd-javascript/5.8.0/pmd-javascript-5.8.0.pom
Downloading: https://oss.sonatype.org/content/repositories/codehaus-snapshots/net/sourceforge/pmd/pmd-javascript/5.8.0/pmd-javascript-5.8.0.pom
Downloading: http://sevntu-checkstyle.github.io/sevntu.checkstyle/maven2/net/sourceforge/pmd/pmd-javascript/5.8.0/pmd-javascript-5.8.0.pom
Downloading: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-javascript/5.8.0/pmd-javascript-5.8.0.pom
Downloaded: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-javascript/5.8.0/pmd-javascript-5.8.0.pom (5 KB at 187.6 KB/sec)
Downloading: http://mvn.reveredata.com/maven/net/sourceforge/pmd/pmd-jsp/5.8.0/pmd-jsp-5.8.0.pom
Downloading: https://oss.sonatype.org/content/repositories/codehaus-snapshots/net/sourceforge/pmd/pmd-jsp/5.8.0/pmd-jsp-5.8.0.pom
Downloading: http://sevntu-checkstyle.github.io/sevntu.checkstyle/maven2/net/sourceforge/pmd/pmd-jsp/5.8.0/pmd-jsp-5.8.0.pom
Downloading: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-jsp/5.8.0/pmd-jsp-5.8.0.pom
Downloaded: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-jsp/5.8.0/pmd-jsp-5.8.0.pom (5 KB at 189.2 KB/sec)
Downloading: http://mvn.reveredata.com/maven/net/sourceforge/pmd/pmd-java/5.8.0/pmd-java-5.8.0.jar
Downloading: http://mvn.reveredata.com/maven/net/sourceforge/pmd/pmd-core/5.8.0/pmd-core-5.8.0.jar
Downloading: http://mvn.reveredata.com/maven/net/sourceforge/pmd/pmd-javascript/5.8.0/pmd-javascript-5.8.0.jar
Downloading: http://mvn.reveredata.com/maven/net/sourceforge/pmd/pmd-jsp/5.8.0/pmd-jsp-5.8.0.jar
Downloading: https://oss.sonatype.org/content/repositories/codehaus-snapshots/net/sourceforge/pmd/pmd-core/5.8.0/pmd-core-5.8.0.jar
Downloading: https://oss.sonatype.org/content/repositories/codehaus-snapshots/net/sourceforge/pmd/pmd-javascript/5.8.0/pmd-javascript-5.8.0.jar
Downloading: https://oss.sonatype.org/content/repositories/codehaus-snapshots/net/sourceforge/pmd/pmd-jsp/5.8.0/pmd-jsp-5.8.0.jar
Downloading: https://oss.sonatype.org/content/repositories/codehaus-snapshots/net/sourceforge/pmd/pmd-java/5.8.0/pmd-java-5.8.0.jar
Downloading: http://sevntu-checkstyle.github.io/sevntu.checkstyle/maven2/net/sourceforge/pmd/pmd-core/5.8.0/pmd-core-5.8.0.jar
Downloading: http://sevntu-checkstyle.github.io/sevntu.checkstyle/maven2/net/sourceforge/pmd/pmd-javascript/5.8.0/pmd-javascript-5.8.0.jar
Downloading: http://sevntu-checkstyle.github.io/sevntu.checkstyle/maven2/net/sourceforge/pmd/pmd-java/5.8.0/pmd-java-5.8.0.jar
Downloading: http://sevntu-checkstyle.github.io/sevntu.checkstyle/maven2/net/sourceforge/pmd/pmd-jsp/5.8.0/pmd-jsp-5.8.0.jar
Downloading: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-java/5.8.0/pmd-java-5.8.0.jar
Downloading: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-core/5.8.0/pmd-core-5.8.0.jar
Downloading: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-javascript/5.8.0/pmd-javascript-5.8.0.jar
Downloading: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-jsp/5.8.0/pmd-jsp-5.8.0.jar
Downloaded: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-jsp/5.8.0/pmd-jsp-5.8.0.jar (93 KB at 894.4 KB/sec)
Downloaded: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-javascript/5.8.0/pmd-javascript-5.8.0.jar (125 KB at 843.1 KB/sec)
Downloaded: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-java/5.8.0/pmd-java-5.8.0.jar (796 KB at 4940.2 KB/sec)
Downloaded: https://repo.maven.apache.org/maven2/net/sourceforge/pmd/pmd-core/5.8.0/pmd-core-5.8.0.jar (704 KB at 4313.3 KB/sec)
[WARNING] There are 1 PMD processing errors:
[WARNING] /home/rivanov/java/git-others/checkstyle/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java: Error while processing /home/rivanov/java/git-others/checkstyle/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java
[INFO] 
[INFO] <<< maven-pmd-plugin:3.8:check (default-cli) < :pmd @ checkstyle <<<
[INFO] 
[INFO] --- maven-pmd-plugin:3.8:check (default-cli) @ checkstyle ---
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 26.062 s
[INFO] Finished at: 2017-06-29T14:29:21-07:00
[INFO] Final Memory: 56M/1465M
[INFO] ------------------------------------------------------------------------
```

It is SUCCESS, but look at output to line 
`
[WARNING] There are 1 PMD processing errors:
[WARNING] /home/rivanov/java/git-others/checkstyle/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java: Error while processing /home/rivanov/java/git-others/checkstyle/checkstyle/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java
`

**Comment by jsotuyod** [comment]
@romani interesting, it doesn't show up on the report (these usually do, under a `<error>` tag in the xml report).

I'll investigate.

**Comment by jsotuyod** [comment]
This was caused by the same bug as #471, the current pmd/5.8.x head works.

Expect release 5.8.1 soon. I'll track the error issue on #474
